src/domain/discounts/details/index.js

```diff
@@ -338,10 +339,46 @@ const DiscountDetails = ({ id }) => {
             </Box>
           </Box>
           <Divider m={3} />
-          <Box>
-            <Text ml={3} mb={2}>
-              Applicable for all products
-            </Text>
+
+          <Box px={3}>
+            {discount.rule.allocation !== "total" &&
```

Comment:
See above comment. Applicable products should be available regardless of allocation method.